domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -922,6 +933,54 @@ public void save(AsyncResult<SQLConnection> sqlConnection, String table, String
     }
   }
 
+  /**
+   * Save entity in table and return the updated entity.
+   *
+   * @param sqlConnection connection (for example with transaction)
+   * @param table where to insert the entity record
+   * @param id  the value for the id field (primary key)
+   * @param entity  the record to insert, a POJO
+   * @param replyHandler  where to report success status and the entity after applying any database INSERT triggers
+   */
+  private <T> void saveAndReturnUpdatedEntity(AsyncResult<SQLConnection> sqlConnection, String table, String id, T entity,
+      Handler<AsyncResult<T>> replyHandler) {
+
+    log.debug("save (with connection and id) called on " + table);
+
+    if (sqlConnection.failed()) {
+      log.error(sqlConnection.cause().getMessage(), sqlConnection.cause());
+      replyHandler.handle(Future.failedFuture(sqlConnection.cause()));
+      return;
+    }
+
+    try {
+      long start = System.nanoTime();
+      String sql = INSERT_CLAUSE + schemaName + DOT + table
+          + " (id, jsonb) VALUES (?, ?::JSONB) RETURNING jsonb";
+      JsonArray jsonArray = new JsonArray().add(id).add(pojo2json(entity));
+      sqlConnection.result().queryWithParams(sql, jsonArray, query -> {
+          statsTracker(SAVE_STAT_METHOD, table, start);
+          if (query.failed()) {
+            log.error(query.cause().getMessage(), query.cause());
+            replyHandler.handle(Future.failedFuture(query.cause()));
+            return;
+          }
+          try {
+            String updatedEntityString = query.result().getResults().get(0).getValue(0).toString();
+            @SuppressWarnings("unchecked")
+            T updatedEntity = (T) mapper.readValue(updatedEntityString, entity.getClass());
+            replyHandler.handle(Future.succeededFuture(updatedEntity));
+          } catch (Exception e) {
+            log.error(e.getCause().getMessage(), e.getCause());
```

Comment:
There is no code coverage for this line because e.getCause() is always null and throws an NullPointerException. Please fix.